eks/nomad/main.tf

```diff
@@ -15,11 +15,12 @@ resource "aws_key_pair" "ssh_key" {
 module "nomad_tls" {
   source   = "../../shared/modules/tls"
   basename = var.basename
+  enabled  = 1 # It can be dangerous to modify this value
```

Comment:
I think you can save a lot of work using [count](https://www.terraform.io/docs/configuration/meta-arguments/count.html) here and maybe use a variable that can be conveniently set via `terraform.tfvars` or cli on the top level. Then you can leave the TLS module as-is.

-- same applies to GKE